kratos/containers/variable.h

```diff
@@ -168,15 +168,9 @@ class Variable : public VariableData
     ///@{
 
     /**
-     * @brief Assignment operator.
-     * @param rOtherVariable The old variable to be assigned
+     * @brief Assignment operator, deleted to avoid misuse which can lead to memory problems
      */
-    VariableType& operator=(const VariableType& rOtherVariable)
-    {
-        VariableData::operator=(rOtherVariable);
-        mZero = rOtherVariable.mZero;
```

Comment:
btw this implementation was incomplete in any case, the time derivative variable is missing :D